src/pages/resultsView/plots/LastPlotsTabSelectionForDatatype.spec.ts

```diff
@@ -0,0 +1,214 @@
+import {assert} from 'chai';
+import LastPlotsTabSelectionForDatatype from './LastPlotsTabSelectionForDatatype';
+import { AxisMenuSelection, MutationCountBy } from './PlotsTab';
+
+// couldn't figure out how to do this with sinon
+function createFakeCallbacks() {
+    const geneCalls: any[] = [];
+    const genesetCalls: any[] = [];
+    const sourceCalls: any[] = [];
+    const treatmentCalls: any[] = [];
+
+    return {
```

Comment:
you should indeed use sinon for this.   you use stubs.  
const someStub = new sinon.stub();  
assert.equal(someStub.args.length, 1) // args is an array of length corresponding to number of times stub was invoked.   each element should be the array of parameters passed to function when it was invoked

Reply:
sinon also has something called "fakes" which might make this even easier, but we would have to upgrade to 5.0.  not necessary

Reply:
ah, thanks